packages/fxa-auth-server/test/local/routes/subscriptions.js

```diff
@@ -2319,7 +2319,92 @@ describe('DirectStripeRoutes', () => {
     });
   });
 
-  describe('updatePayment', () => {});
+  describe('updatePayment', () => {
+    let request;
+    const paymentToken = 'tok_visa';
+
+    beforeEach(() => {
+      request = Object.create(VALID_REQUEST);
+      request.payload = { paymentToken };
+
+      directStripeRoutesInstance.stripeHelper.updateCustomerPaymentMethod.returns();
+    });
+
+    describe('when the customer exists', () => {
+      it('updates the payment method', async () => {
+        const customer = Object.create(customerFixture);
+        directStripeRoutesInstance.stripeHelper.fetchCustomer.returns(customer);
+
+        await directStripeRoutesInstance.updatePayment(request);
+
+        assert.isTrue(
+          directStripeRoutesInstance.log.begin.calledOnceWith(
+            'subscriptions.updatePayment',
+            request
+          )
+        );
+
+        assert.isTrue(
+          directStripeRoutesInstance.customs.check.calledOnceWith(
+            request,
+            TEST_EMAIL,
+            'updatePayment'
+          )
+        );
+
+        assert.isTrue(
+          directStripeRoutesInstance.stripeHelper.updateCustomerPaymentMethod.calledOnceWith(
+            customer.id,
+            paymentToken
+          )
+        );
+
+        assert.isTrue(
+          directStripeRoutesInstance.log.info.calledOnceWith(
+            'subscriptions.updatePayment.success',
+            { uid: UID }
+          )
+        );
+      });
+    });
+
+    describe('when the customer does not exist', () => {
+      it('throws an error', () => {
+        directStripeRoutesInstance.stripeHelper.fetchCustomer.returns();
+
+        return directStripeRoutesInstance.updatePayment(request).then(
```

Comment:
Mixing `async/await` in the previous test and promises here is kind of weird.  Can you also using `async/await` here, and `try/catch` the error you want to test against?

Reply:
I can do that. I saw errors tested different ways in the code base when it came to promises - `try/catch` and `.then(()=>{}, err=>{})`. It didn't seem that we had a preferred methodology for error testing.